lib/install-go-binary.js

```diff
@@ -113,26 +122,28 @@ async function installBinary( { arch = process.arch, platform = process.platform
 
 	let fd;
 	try {
+		// eslint-disable-next-line security/detect-non-literal-fs-filename
 		fd = await fs.promises.open( writeTo, 'w', 0o755 );
 	} catch ( err ) {
 		return Promise.reject( `Could not open the destination file for writing: ${ err }` );
 	}
 
 	debug( 'OK to write.' );
 
-	return new Promise( async ( resolve, reject ) => {
-		let responseStream;
-		try {
-			responseStream = await downloadBinary( { platform, arch } );
-		} catch ( downloadErr ) {
-			debug( { downloadErr } );
-			return reject( downloadErr );
-		}
+	let responseStream;
+	try {
+		responseStream = await downloadBinary( { platform, arch } );
+	} catch ( downloadErr ) {
+		debug( { downloadErr } );
+		return Promise.reject( downloadErr );
+	}
```

Comment:
This is because of the [`no-async-promise-executor` rule](https://eslint.org/docs/latest/rules/no-async-promise-executor). The suggestion was to reduce the scope of the `Promise` constructor, which worked.